crates/bevy_ecs/src/component.rs

```diff
@@ -231,12 +271,13 @@ pub struct ComponentId(usize);
 
 impl ComponentId {
     #[inline]
-    pub const fn new(index: usize) -> ComponentId {
```

Comment:
The primary one is to avoid users creating their own `ComponentId`s arbitrarily and converting them into something that might be saved for the exact same reasons you've enumerated: it's not stable across runs and shouldn't be easy to convert to something where you can easily manipulate the representation and reconstruct a valid ComponentId out of it, nor allow users tot serialize it to a file or over the network.

Is the use case here to communicate this across runtime boundaries (i.e. scripting)?

Reply:
Yes, we need to be able to pass component IDs to and from scripts to query components from an entity.

Reply:
If the point is to cross a potential FFI boundary for scripting, we can set it to `repr(transparent)` if need be, is that sufficient?